src/bundle.ts

```diff
@@ -0,0 +1,32 @@
+export default class Bundle {
+	public tarStream: NodeJS.ReadableStream
+
+	/**
+	 * deviceType: The slug of the device type that this bundle has been created
+	 * for
+	 */
+	public deviceType: string
+
+	/**
+	 * architecture: The architecture that this resin bundle is targeting
+	 */
+	public architecture: string
```

Comment:
That was my thoughts as well. These modules are being separated out in the hopes that a user can deploy them in any way they see fit, so tying them just to architectures and devices which resin supports doesn't make sense IMO. That said, utilising typescript's compile time protection in regards to this would be pretty sweet, I'm just not sure how general we could make it.